src/test/scala/org/camunda/feel/impl/builtin/BuiltinTemporalFunctionsTest.scala

```diff
@@ -46,4 +51,38 @@ class BuiltinTemporalFunctionsTest
       eval(""" today() """) should be(ValDate(now.toLocalDate))
   }
 
+  "The day of year() function" should "return the day within the year" in {
+
+    eval(s"day of year($date)") should be(ValNumber(260))
+    eval(s"day of year($localDateTime) ") should be(ValNumber(260))
+    eval(s"day of year($dateTime) ") should be(ValNumber(260))
```

Comment:
Considering the edge cases for `week of year`. Perhaps it makes sense to add some edge cases here too. Years with more or less than 365 days (if that exists).

Reply:
Also, can we assume that a `date/datetime` object is always a real date. For example does `date(2020,9,40)` become `null` because it could not be parsed, or does it become a real date object?

Reply:
Yes. It would fail to create this date and return `null` instead.